utxonursery.go

```diff
@@ -964,101 +940,223 @@ func deserializeKidList(r io.Reader) ([]*kidOutput, error) {
+// all goes well, and the timeout transaction is successfully confirmed, the
+// the now-mature kidOutput will be unwrapped and continue its journey through
+// the nursery.
+type babyOutput struct {
+	kidOutput
+
+	expiry    uint32
+	timeoutTx *wire.MsgTx
+}
+
+// makeBabyOutput constructs baby output the wraps a future kidOutput. The
+// provided sign descriptors and witness types will be used once the output
+// reaches the delay and claim stage.
+func makeBabyOutput(outpoint, originChanPoint *wire.OutPoint,
+	blocksToMaturity uint32, witnessType lnwallet.WitnessType,
+	htlcResolution *lnwallet.OutgoingHtlcResolution) babyOutput {
+
+	kid := makeKidOutput(outpoint, originChanPoint,
+		blocksToMaturity, witnessType,
+		&htlcResolution.SweepSignDesc)
+
+	return babyOutput{
+		kidOutput: kid,
+		expiry:    htlcResolution.Expiry,
+		timeoutTx: htlcResolution.SignedTimeoutTx,
+	}
+}
+
+// Encode writes the baby output to the given io.Writer.
+func (bo *babyOutput) Encode(w io.Writer) error {
+	var scratch [4]byte
+	byteOrder.PutUint32(scratch[:], bo.expiry)
+	if _, err := w.Write(scratch[:]); err != nil {
+		return err
+	}
+
+	if err := bo.timeoutTx.Serialize(w); err != nil {
+		return err
+	}
+
+	return bo.kidOutput.Encode(w)
+}
+
+// Decode reconstructs a baby output using the provide io.Reader.
+func (bo *babyOutput) Decode(r io.Reader) error {
+	var scratch [4]byte
+	if _, err := r.Read(scratch[:]); err != nil {
+		return err
+	}
+	bo.expiry = byteOrder.Uint32(scratch[:])
+
+	bo.timeoutTx = new(wire.MsgTx)
+	if err := bo.timeoutTx.Deserialize(r); err != nil {
+		return err
+	}
+
+	return bo.kidOutput.Decode(r)
+}
+
+// kidOutput represents an output that's waiting for a required blockheight
+// before its funds will be available to be moved into the user's wallet.  The
+// struct includes a WitnessGenerator closure which will be used to generate
+// the witness required to sweep the output once it's mature.
+//
+// TODO(roasbeef): rename to immatureOutput?
+type kidOutput struct {
+	breachedOutput
+
+	originChanPoint wire.OutPoint
+
+	// TODO(roasbeef): using block timeouts everywhere currently, will need
+	// to modify logic later to account for MTP based timeouts.
+	blocksToMaturity uint32
+	confHeight       uint32
+}
+
+func makeKidOutput(outpoint, originChanPoint *wire.OutPoint,
+	blocksToMaturity uint32, witnessType lnwallet.WitnessType,
+	signDescriptor *lnwallet.SignDescriptor) kidOutput {
+
+	return kidOutput{
+		breachedOutput: makeBreachedOutput(
+			outpoint, witnessType, signDescriptor,
+		),
+		originChanPoint:  *originChanPoint,
+		blocksToMaturity: blocksToMaturity,
+	}
+}
+
+func (k *kidOutput) OriginChanPoint() *wire.OutPoint {
+	return &k.originChanPoint
+}
+
+func (k *kidOutput) BlocksToMaturity() uint32 {
+	return k.blocksToMaturity
+}
+
+func (k *kidOutput) SetConfHeight(height uint32) {
+	k.confHeight = height
+}
+
+func (k *kidOutput) ConfHeight() uint32 {
+	return k.confHeight
+}
+
```

Comment:
We can add the following fragment to insert a compile-time constant asserting that `kidOutput` meets the `CsvSpendableOutput` interface: 
```go
var _ CsvSpendableOutput = (*kidOutput)(nil)
```